tensorflow/tools/xla_extract/tf_graph_to_xla_lib.cc

```diff
@@ -137,25 +137,39 @@ xla::HloModuleProto ExtractHloFromGraphDef(const GraphDef& in_graph,
     LOG(INFO) << "cluster not found, using "<<fdef.signature().name()<<" instead\n";
   }
 
-  // Need to rearrange these xla_args to match graph.
-  // features and labels should be the first nodes but sometimes aren't
-  // If args are [a, b, c, d, e, ...] and a is the features arg, then leave
-  // as is.  if d is the features arg, then reorder as [d, e, ..., a, b, c]
-  // This corresponds to a rotation with `d` as the axis.
   auto xla_args = BuildXlaArgsFromClientGraph(client_graph);
 
-  // Find the features arg (first parameter tensor)
-  auto features_iter = std::find_if(
-      xla_args.begin(), xla_args.end(),
-      [](const XlaCompiler::Argument& xarg) -> bool {
-        return (xarg.kind == XlaCompiler::Argument::kParameter &&
-                xarg.resource_kind == 0 && xarg.initialized == false);
-      });
 
-  // Rotate around it
-  std::rotate(xla_args.begin(), features_iter, xla_args.end());
+  // to make sure xla_args matches fdef
+  std::vector<XlaCompiler::Argument> new_xla_args;
+  auto fdef_ground = fdef.signature().input_arg();
+
+  for(int i = 0;i<xla_args.size();i++){
+    std::string xla_arg_name=xla_args[i].name;
```

Comment:
for all of these string transformations, please try to use the already provided utilities like:
```
File: tensorflow/tensorflow/core/lib/strings/str_util.h
78: string Lowercase(StringPiece s);

92: string ArgDefCase(StringPiece s);
``` 
etc...

Reply:
oh didnt know about these. will swap them out :)